### PR TITLE
Fix build

### PR DESCRIPTION
**Pull request by swallez**
This updates the build system to handle the recent changes in Elasticsearch:
- rest api specs and yaml test definitions are now downloadable artifacts available at artifacts.elastic.co
- ES security is on by default, meaning we have to use authentication also for tests of the free tier APIs.

Client code has also been updated to the latest 8.0.0-SNAPSHOT APIs.

The code that downloads artifacts is in a new `xtask` package that follows the [cargo-xtasks conventions](https://github.com/matklad/cargo-xtask), derived from the ["Make your own make"](https://matklad.github.io/2018/01/03/make-your-own-make.html) blog post. The logic to download artifacts is complex enough that it doesn't fit as a script in `Make.toml`, and is therefore implemented as a CLI tool that is exposed as a task in `Make.toml`.

Note: utimately we may consider refactoring all of `Make.toml` to `xtask`. `cargo-make` is a nice platform-independent alternative to `make` but it comes with its own weight and as the makefile grows it starts to be hard to read and navigate. Transitioning to "real code" with the help of a few crates that ease external process management may actually be more maintainable.

### CODE REVIEW

**Review by russcam — APPROVED** [review]
LGTM 👍 

I left a suggestion for a small typo.

Also looks like api_generator/docs/namespaces needs fleet.md and shutdown.md doc files

```
2021-09-06 09:40:40,074 WARN  [api_generator::generator::output] Missing docs file Some("./api_generator/docs\\namespaces\\fleet.md")
2021-09-06 09:40:40,084 WARN  [api_generator::generator::output] Missing docs file Some("./api_generator/docs\\namespaces\\shutdown.md")
```

File: api_generator/Cargo.toml [hunk 1]
```diff
@@ -9,9 +9,9 @@ edition = "2018"
 license = "Apache-2.0"
 
 [dependencies]
+anyhow = "1.0"
```
[the remark this answers is not in the document]
Comment (swallez) [reply api_generator/Cargo.toml:12]: [Failure has been deprecated](https://github.com/rust-lang-nursery/failure) and now suggests `anyhow` as a replacement. They are somehow similar, although I understand it `anyhow` uses the recent changes in the sdtlib's `Error` trait. But now that `failure` is deprecated we have to migrate away from it.

File: Makefile.toml [hunk 2]
```diff
@@ -25,6 +25,16 @@ private = true
 condition = { env = { "TEST_SUITE" = "platinum" }, env_not_set = ["ELASTICSEARCH_URL"] }
 env = { "ELASTICSEARCH_URL" = "https://elastic:changeme@localhost:9200" }
 
+[tasks.download-specs]
+category = "Elasticsearch"
+description = '''Download Rest API specs and YAML tests'''
+private = true
+command = "cargo"
+# cargo-make insists on installing cargo-xtask and ignores .cargo/config.toml
```
Comment (sagiegurari) [inline Makefile.toml:33]: @swallez you can disable it by adding:
install_crate = false
see more details at:
https://github.com/sagiegurari/cargo-make#cargo-plugins

Comment (swallez) [reply Makefile.toml:33]: Thanks for the hint @sagiegurari!

**Pull request merged**